### PR TITLE
Ignore errors during logging

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

This prevent loops when errors happen during error logging, catch them.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by silamon — APPROVED** [review]

**Comment by silamon** [comment]
While the errors aren't being logged to backend anymore, the loop still happens silently I guess. 
I think these issues are relevant #17889 and most likely #17885.

**Comment by silamon** [comment]
> > While the errors aren't being logged to backend anymore, the loop still happens silently I guess.
> 
> It should not, the error logging itself will not trigger the error logging as we now catch the errors during error logging

Yeah, understood, the current stable still has the loop while this pull request will solve that. 👍

File: src/state/logging-mixin.ts [hunk 1]
```diff
@@ -40,30 +40,46 @@ export const loggingMixin = <T extends Constructor<HassBaseEl>>(
...
-        });
+          );
+          await this._writeLog({
+            // The error object from browsers includes the message and a stack trace,
+            // so use the data in the error event just as fallback
+            message,
+          });
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error("Error during logging error:", message, e);
```
Comment (steverep) [inline src/state/logging-mixin.ts:59]: Don't need `message` here as it would already be in the console.

**Review by steverep — COMMENTED** [review]
@bramkragten why isn't the connection check stopping the process before it has a chance to generate another error?

I think the try/catch is ultimately needed to prevent loops, but I'm trying to understand how it gets in this situation in the first place.

**Pull request merged**